### PR TITLE
Syntax updates to switch and support classes

### PR DESCRIPTION
**Pull request by lcampos**
This PR fixes tokenization issue with switch statements with multiple string conditions. It also addresses an issue with tokens being incorrectly assigned to methods, variables, etc when they contain a support class name e.g. SObjectUtility.

**Before**
Screen Shot 2020-02-08 at 11 55 22 AM

Screen Shot 2020-02-10 at 5 55 37 PM

**After**
Screen Shot 2020-02-10 at 5 56 53 PM

Screen Shot 2020-02-10 at 5 56 24 PM

@W-6532166@

### CODE REVIEW

**Comment by brpowell** [comment]
Screen Shot 2020-02-11 at 3 13 41 PM

It looks like we're still missing the `string.quoted.single.apex` scope on the when statement strings

File: src/apex.tmLanguage.yml [hunk 1]
```diff
@@ -946,18 +946,28 @@ repository:
...
+  
+  when-string-statement:
+    patterns:
+    - name: string.quoted.single.apex
+      begin: \'
+      beginCaptures:
+        '0': { name: punctuation.definition.string.begin.apex }
+      end: \'
+      endCaptures:
+        '0': { name: punctuation.definition.string.end.apex }
```
Comment (brpowell) [inline src/apex.tmLanguage.yml]: I think you might actually be able to remove these "when-string-*" specific scopes and just comment out [this line](https://github.com/forcedotcom/apex-tmLanguage/pull/30/files#diff-efe8ffbdf717bb7edbba667d56c26053L927) to fix the issue.

The "when-statement" scope already attempts to match anything in the "expression" scope which includes string literals and allows you to not repeat work of trying to match them. The exact path is `switch-statement -> when-statement -> expression -> literal -> string-literal`.

Let me know if you see anything wrong with this approach.

Comment (lcampos) [reply src/apex.tmLanguage.yml]: Let me try, I don't exactly remember why I coded the switch this way but I'm ok with simplifying it

Comment (lcampos) [reply src/apex.tmLanguage.yml]: We synced offline, the change to simply switch statements causes a couple of test failures so we'll leave this as is for now.

**Pull request merged**